### PR TITLE
Refactor `Registration` and `WebinarTemplate`

### PR DESCRIPTION
**Pull request by sandylcruz**
# Summary
* Refactored `Registration` and `WebinarTemplate`  (now `Webinar`) into multiple smaller components
* Refactor styling to avoid unwanted layout shift when loading Marketo form in
* Added `webinarFull` and `registrationFull` prop to their corresponding components.

# Previews
* Preview `Registration`:
     * [/fenway-park-signup](https://next-git-sandy-registration-gravitational.vercel.app/fenway-park-signup/)
     * [/kubecon-happy-hour](https://next-git-sandy-registration-gravitational.vercel.app/kubecon-happy-hour/)
     * [/newsletter](https://next-git-sandy-registration-gravitational.vercel.app/newsletter/)
     * [/rsa-happy-hour](https://next-git-sandy-registration-gravitational.vercel.app/rsa-happy-hour/)
     * [/rsa-lunch-enterprise](https://next-git-sandy-registration-gravitational.vercel.app/rsa-lunch-enterprise/)
     * [/rsa-lunch-md-market](https://next-git-sandy-registration-gravitational.vercel.app/rsa-lunch-mid-market/)
     * [/topgolf-event](https://next-git-sandy-registration-gravitational.vercel.app/topgolf-event/)
* Preview `Webinar` 
     * [/identity-based...](https://next-git-sandy-registration-gravitational.vercel.app/identity-based-data-security-on-aws-webinar/) 
     * [/introduction-to-teleport](https://next-git-sandy-registration-gravitational.vercel.app/introduction-to-teleport/)
     * [/using-teleport-to-secure...](https://next-git-sandy-registration-gravitational.vercel.app/using-teleport-to-secure-ssh-and-kubernetes-access/)

# Use of `content` in `webinarFull` and `registrationFull`
* `jsx` can be passed to `Registration` and `Webinar` as an optional prop, as shown below:

    ```
  <Registration
        formID="1294"
        headTitle="KubeCon"
        headDescription="Register"
        title="AQUA + TELEPORT"
        mainImage={teleportAqua}
        description={eventDescription}
        CTA="Join the Party!"
        subCTA="Join us for a VIP Cocktail Party following KubeCon on Thursday, May 19!"
        dark
        secondLogo={aquaLogo}
        imageContainerHeight={[254, 300]}
        previewImage="og-cards/events/kubecon-happy-hour.jpg"
        registrationFull={{
          show: true,
          content: (
            <>
              <h1>Lorem ipsum title</h1>
              <p>Lorem ipsum description</p>
            </>
          ),
        }}
      />
     ``` 
Screen Shot 2022-05-11 at 4 16 16 PM

# Default shown if `content` is not provided
Screen Shot 2022-05-10 at 9 45 54 AM
Screen Shot 2022-05-10 at 9 46 23 AM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202247873317569

### CODE REVIEW

File: archive/teleport-9-webinar/index.mdx [hunk 1]
```diff
@@ -7,14 +7,14 @@ emailSubscribe: false
...
+import { Webinar } from "components/Webinar";
 import PromoPointsBlock from "components/PromoPointsBlock";
 import hero from "./assets/machine-id.png";
 import speakerImage from "./assets/bothSpeakers@2x.png";
 
-<WebinarTemplate
+<Webinar
   title="Teleport 9: Identity for All Your Machines and a Lot More"
-  formId="1273"
+  formID="1273"
```
Comment (jeffgaynor) [inline archive/teleport-9-webinar/index.mdx:17]: `formId` is standard, should remain in camel case everywhere.

File: components/Registration/Registration.tsx [hunk 2]
```diff
@@ -42,6 +39,7 @@ export const Registration = ({
   dark = false,
   previewImage,
   imageContainerHeight = [158, 300],
+  registrationFull = { show: false },
```
Comment (jeffgaynor) [inline components/Registration/Registration.tsx:42]: Can remove this default as it's set inside RegistrationForm component.

File: components/Registration/RegistrationForm.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    boxSizing: "border-box",
+    display: "flex",
+    justifyContent: "center",
+    minWidth: "432px",
+  })
+);
+
+export const RegistrationForm = ({
+  formID,
+  registrationFull = { show: false },
```
Comment (jeffgaynor) [inline components/Registration/RegistrationForm.tsx:33]: Can remove the default and set inside component.

File: components/Registration/RegistrationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  CTA,
+  subCTA,
+}: RegistrationFormProps) => {
+  const { show, content } = registrationFull;
```
Comment (jeffgaynor) [inline components/Registration/RegistrationForm.tsx:37]: Shorter code to move just to this spot:
```ts
const { show = false, content } = registrationFull;
```

File: pages/kubecon-happy-hour/index.tsx [hunk 5]
```diff
@@ -13,7 +13,7 @@ const StyledP = styled("p")(
 
 const eventDescription = (
   <>
-    VIP Cocktail Party at KubeCon + CloudNativeCon
+    KubeCon VIP Cocktail Party
```
Comment (jeffgaynor) [inline pages/kubecon-happy-hour/index.tsx:16]: Was dropping CouldNativeCon requested? It still includes that on prod, and I see it was added last week: https://app.asana.com/0/1201922182803749/1202239342965559/f

**Review by jeffgaynor — CHANGES_REQUESTED** [review]
Nice changes! A couple of small notes... please confirm copy change for KubeCon event is valid.

File: archive/teleport-9-webinar/index.mdx (hunk 1 above)
Comment (sandylcruz) [reply archive/teleport-9-webinar/index.mdx:17]: Updating in both `Webinar` and `Registration`

File: pages/kubecon-happy-hour/index.tsx (hunk 5 above)
Comment (sandylcruz) [reply pages/kubecon-happy-hour/index.tsx:16]: No, I reset my branch without rebasing to make sure updates were all there. Updating

File: components/Registration/RegistrationForm.tsx (hunk 3 above)
Comment (sandylcruz) [reply components/Registration/RegistrationForm.tsx:33]: Yes, updating

File: components/Registration/Registration.tsx (hunk 2 above)
Comment (sandylcruz) [reply components/Registration/Registration.tsx:42]: If removed, it gives a build error: 
```
Error occurred prerendering page "/kubecon-happy-hour". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read property 'show' of undefined
    at RegistrationForm (/Users/sandycruz/Code/teleport/next/.next/server/chunks/3708.js:165:5)
    at d (/Users/sandycruz/Code/teleport/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:33:498)
    at bb (/Users/sandycruz/Code/teleport/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:36:16)
    at a.b.render (/Users/sandycruz/Code/teleport/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:42:43)
    at a.b.read (/Users/sandycruz/Code/teleport/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:41:83)
    at Object.exports.renderToString (/Users/sandycruz/Code/teleport/next/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:52:138)
    at renderPage (/Users/sandycruz/Code/teleport/next/node_modules/next/dist/server/render.js:751:45)
    at Object.ctx.renderPage (/Users/sandycruz/Code/teleport/next/.next/server/pages/_document.js:37:30)
    at Object.defaultGetInitialProps (/Users/sandycruz/Code/teleport/next/node_modules/next/dist/server/render.js:389:51)
    at Function.getInitialProps (/Users/sandycruz/Code/teleport/next/.next/server/chunks/6859.js:618:16)
```

Comment (jeffgaynor) [reply components/Registration/Registration.tsx:41]: Did you add it in the other location I mentioned first? I tried it locally and it seems to work.

Comment (sandylcruz) [reply components/Registration/Registration.tsx]: The problem I was hitting was that `registrationFull` was potentially undefined, as it's an optional prop. Updating to handle that case now.

**Review by jeffgaynor — APPROVED** [review]
🚀 🥇 Looks good!

**Pull request merged**